### PR TITLE
Sanitize video ID input

### PR DESCRIPTION
**Pull request by alexbox23**
Parse for valid youtube slugs and URLs.
https://github.com/elldritch/jukebox/issues/1

### CODE REVIEW

File: web/src/Room.tsx [hunk 1]
```diff
@@ -117,6 +117,37 @@ export default function Room() {
...
+
+    // https://stackoverflow.com/questions/19377262/regex-for-youtube-url
+    // See https://regex101.com/r/vHEc61/1 for test cases.
+    const ytRegex = new RegExp(
+      /^((?:https?:)?\/\/)?/.source + /((?:www|m)\.)?/.source
+      + /((?:youtube(?:-nocookie)?\.com|youtu.be))/.source
+      + /(\/(?:[\w\-]+\?v=|embed\/|live\/|v\/)?)/.source
+      + /([\w\-]+)/.source // The slug is contained in the 5th group
+      + /(\S+)?$/.source
+    );
```
Comment (elldritch) [inline web/src/Room.tsx]: Don't use a regular expression for this. Instead, use the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) API. You should be able to use it to parse the URL and extract the query parameter.

File: web/src/Room.tsx [hunk 2]
```diff
@@ -516,9 +547,10 @@ export default function Room() {
             className="flex rounded-md shadow-sm"
             onSubmit={(e) => {
               e.preventDefault();
+              const slug = sanitizeSlug(addToQueueInput);
```
Comment (elldritch) [inline web/src/Room.tsx]: It's only ever used once, and it doesn't have independent tests, so I'd probably just inline this.

**Review by elldritch — CHANGES_REQUESTED** [review]

File: web/src/Room.tsx (hunk 2 above)
Comment (alexbox23) [reply web/src/Room.tsx]: Went a bit overboard learning new syntax LOL

File: web/src/Room.tsx (hunk 1 above)
Comment (alexbox23) [reply web/src/Room.tsx]: Ah much cleaner now. Lmk if this looks right.

**Pull request merged**